src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -19,7 +23,32 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        TreeSet<Character> charSet = new TreeSet<>();
+        StringBuilder charBuilder = new StringBuilder();
+        int iterator = 5;
+        File file = new File(fileName);
+        if (!file.exists()) {
+            throw new FileNotFoundException("File doesn't exist.");
+        }
+        try (FileInputStream fileInputStream = new FileInputStream(file)) {
+            while (fileInputStream.available() > 0) {
+                char alphabetChar = (char) fileInputStream.read();
+                if (Character.isLetter(alphabetChar)) {
+                    charSet.add(Character.toLowerCase(alphabetChar));
+                }
+            }
+            for (Character nextCharacter : charSet) {
+                charBuilder.append(nextCharacter);
+                iterator++;
+                if (iterator == 5) {
```

Comment:
better to do on start of loop smth like that:
`if (charBuilder.length() == MAX_LENGTH) {
break;
}`
And return charBuilder.toString();
As result you don;t need to do extra operations of substrings by condition

Reply:
@al-pavlenko don